website_cms/models/cms_media.py

```diff
@@ -0,0 +1,313 @@
+# -*- coding: utf-8 -*-
+
+# pylint: disable=E0401
+# pylint: disable=W0212
+from openerp import models
+from openerp import fields
+from openerp import api
+# from openerp.tools.translate import _
+from openerp.addons.website.models.website import slug
+
+from openerp.addons.website_cms.utils import IMAGE_TYPES
+from openerp.addons.website_cms.utils import VIDEO_TYPES
+from openerp.addons.website_cms.utils import AUDIO_TYPES
+from openerp.addons.website_cms.utils import download_image_from_url
+from openerp.addons.website_cms.utils import guess_mimetype
+from openerp.addons.website_cms.utils import AttrDict
+
+import urlparse
+
+
+# XXX: we should move this stuff to website_cms
+# and use services like http://noembed.com
+EMBED_PATTERN = AttrDict({
+    'youtube': AttrDict({
+        'key': 'v',
+        'url': 'https://www.youtube.com/embed/{vid}',
+    }),
+})
+IMAGE_PREVIEW_PATTERN = AttrDict({
+    'youtube': AttrDict({
+        'key': 'v',
+        'url': 'https://i.ytimg.com/vi/{vid}/hqdefault.jpg',
+    }),
+})
+
+
+def to_embed_url(url, provider='youtube'):
+    """Simple function to convert video url to embed url.
+
+    Example:
+
+        >>> to_embed_url('https://www.youtube.com/watch?v=Z5cterm_nW0')
+        https://www.youtube.com/embed/Z5cterm_nW0
+
+    The best would be to make use of services like `oembed`
+    but we are taking this easy at the moment ;)
+    """
+    pattern = EMBED_PATTERN.get(provider)
+    parsed = urlparse.urlparse(url)
+    qstring = urlparse.parse_qs(parsed.query)
+    if pattern and pattern.key in qstring:
+        return pattern.url.format(vid=qstring[pattern.key][0])
+    return url
+
+
+def to_preview_url(url, provider='youtube'):
+    """Simple function to convert video url to image preview url.
+
+    Example:
+
+        >>> to_preview_url('https://www.youtube.com/watch?v=Z5cterm_nW0')
+        https://i.ytimg.com/vi/Z5cterm_nW0/hqdefault.jpg
+
+    The best would be to make use of services like `oembed`
+    but we are taking this easy at the moment ;)
+    """
+    pattern = IMAGE_PREVIEW_PATTERN.get(provider)
+    parsed = urlparse.urlparse(url)
+    qstring = urlparse.parse_qs(parsed.query)
+    if pattern and pattern.key in qstring:
+        return pattern.url.format(vid=qstring[pattern.key][0])
+    return url
+
+
+class CMSMedia(models.Model):
+    """Model of a CMS media."""
+
+    _name = 'cms.media'
+    _description = 'CMS Media'
+    _order = 'sequence, id'
+    _inherit = ['ir.attachment',
+                'website.published.mixin',
+                'website.orderable.mixin',
+                'website.coremetadata.mixin',
+                'website.image.mixin',
+                'website.security.mixin']
+
+    name = fields.Char(
+        'Name',
+        required=True,
+        translate=True,
+    )
+    description = fields.Text(
+        'Description',
+        translate=True,
```

Comment:
The "translation" thing is still to be re-thinked: the best would be to get rid of odoo translation mechanism and have separated contents per each language.